app/controllers/admin/categories_controller.rb

```diff
@@ -1,6 +1,9 @@
 class Admin::CategoriesController < Admin::BaseController
   before_action :find_category, only: %i(edit update destroy)
   before_action :check_category_product, only: :destroy
+
+  authorize_resource
```

Comment:
- method này có tác dụng gì em?
- nếu chỉ muốn phân quyền user cho 1 số actions trong controller thì có cách nào ko?
- ngoài method  này thì gem cancancan còn cung cấp cách nào khác để phân quyền cho action?

Reply:
- Method authorize_resource sẽ tự động thực hiện kiểm tra quyền của user với resource trước mỗi action.
- Nếu muốn phân quyền user cho 1 số action, ta thay đổi các quyển của user trong model ability.rb :
alias_action :index, :show, :to => :read
alias_action :new, :to => :create
alias_action :edit, :to => :update
- Ngoài method này, gem cancancan còn có method authorize! để kiểm tra quyền người dùng, 
ngoài ra còn các method load_resource và load_and_authorize_resource
load_resource sẽ tự động thực hiện load resource trước mỗi action (trước khi load nó kiểm tra resource của action tương ứng đã được load chưa, nếu chưa thì sẽ load resource)
load_and_authorize_resource bao gồm load_resource và authorize_resource